### ISSUE TITLE
C-Parser: cant find method definition when using namespaces

### ISSUE THREAD

**Issue by Hanmac** [lsegal/yard#809]
SampleCode

```c++
VALUE rb_cWXRect;

namespace RubyWX {
namespace Rect {

VALUE _alloc(VALUE self)
{
	return wrap(new wxRect());
}

/*
 * call-seq:
 *   inspect -> String
 *
 * Human-readable description.
 * ===Return value
 * String
*/
VALUE _inspect(VALUE self)
{
	return rb_sprintf( "%s(%d, %d, %d, %d)",
		rb_obj_classname( self ),
		FIX2INT(_getX(self)),
		FIX2INT(_getY(self)),
		FIX2INT(_getWidth(self)),
		FIX2INT(_getHeight(self)));
}

}
}

void Init_WXRect()
{
	using namespace RubyWX::Rect;
	rb_cWXRect = rb_define_class("Rect",rb_cObject);

	rb_define_alloc_func(rb_cWXRect,_alloc);

	rb_define_method(rb_cWXRect,"inspect",RUBY_METHOD_FUNC(_inspect),0);

}
```

returns this
```
Files:           2
Modules:         0 (    0 undocumented)
Classes:         1 (    1 undocumented)
Constants:       0 (    0 undocumented)
Methods:         1 (    1 undocumented)
 0.00% documented
```
currenty the Files2 is a bug that happens because one of the o.file is nil
but for my self i cant fix it that it does find the method with the c++ namespaces

**Comment by thomthom** [comment]
I ran into this myself - using C++ to write my Ruby extensions. I've been using namespaces (and anonymous namespaces) instead of static functions in a rather large code base. Tried to use YARD to see if I could use some of the data it collects about the source code - but got stomped by this issue.

I'm not familiar with the YARD code base at all - but I'd be willing to assist in any way possible to allow C++ namespace support.
